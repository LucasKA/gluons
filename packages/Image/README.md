# Mandala UI Image
Mandala is a React component library that utilizes a declarative CSS.

### Description
An image component is used to render a picture element with a `srcset`, and smart fallback. You can pass a string url, or an object with `desktop`, `tablet`, and `mobile` keys. The key values should have an array of one or two image urls, the first for 1x pixel density and one for 2x.

The image Defaults to 100% width regardless of actual image size.

### Installation

Check the [Monorepo README](https://github.com/mandala-ui/mandala) for installation.

### Usage
if using the whole library:
```js
import { Image } from 'mandala';
```
if only using this component:
```js
import Image from '@mandala-ui/image';
```

### Props
* `alt` - the alt text for the image
* `images` - an object that contains the keys `desktop`, `tablet`, and `mobile`, with values having a URL that correlates to the image you want to show on that device.
* `shadow` - drop shadow to add on the image, from 0-5
* `radius` - border radius of the image, from 0-4
* `url` - the URL for the base image. If this is the only thing that is passed, then you don't need to pass an `images` object

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| alt      | string | "blank figure"| Yes          |
| images   | object | null         | -          |
| shadow   | number | 0         | -          |
| radius   | number | 0         | -          |
| url      | string | `http://via.placeholder.com/1600x900/CDECFF`| Yes         |


![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
