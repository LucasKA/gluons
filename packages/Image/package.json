{
  "author": "Lucas K Allmon",
  "description": "Mandala UI Image Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/Image.js",
  "module": "es/Image.js",
  "name": "@mandala-ui/image",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "0.0.1"
}
