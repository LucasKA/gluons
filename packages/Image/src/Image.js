import React from 'react';
import PropTypes from 'prop-types';

const Image = ({
  alt,
  url,
  images,
  radius,
  shadow,
}) => (
  <picture className="image">
    { images && images.desktop ?
      <source
        key={images.desktop[0] || url}
        media="(min-width: 60em)"
        srcSet={`
            ${images.desktop[0] ? images.desktop : null} 1x,
            ${images.desktop[1] ? images.desktop : null} 2x`
          }
      />
        : null
      }
    { images && images.tablet ?
      <source
        key={images.tablet[0] || url}
        media="(min-width: 30em) and (max-width: 60em)"
        srcSet={`
            ${images.tablet[0] ? images.tablet : null} 1x,
            ${images.tablet[1] ? images.tablet : null} 2x`
          }
      />
      : null
    }
    { images && images.mobile ?
      <source
        key={images.mobile[0] || url}
        media="(max-width: 30em)"
        srcSet={`
          ${images.mobile[0] ? images.mobile : null} 1x,
          ${images.mobile[1] ? images.mobile : null} 2x`
        }
      />
    : null
    }
    <img
      className={`br${radius} shadow-${shadow} w-100`}
      src={url}
      alt={alt}
    />
  </picture>
);

Image.propTypes = {
  alt: PropTypes.string,
  url: PropTypes.string,
  images: PropTypes.shape({
    desktop: PropTypes.arrayOf(
      PropTypes.string,
    ),
    mobile: PropTypes.arrayOf(
      PropTypes.string,
    ),
    tablet: PropTypes.arrayOf(
      PropTypes.string,
    ),
  }),
  shadow: PropTypes.number,
  radius: PropTypes.number,
};

Image.defaultProps = {
  alt: 'blank figure',
  url: 'http://via.placeholder.com/1600x900/CDECFF',
  images: null,
  shadow: null,
  radius: 0,
};

export default Image;
