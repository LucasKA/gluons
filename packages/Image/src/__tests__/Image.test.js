import React from 'react';
import { shallow } from 'enzyme';
import Image from '../Image';

describe('<Image />', () => {
  it('exists', () => {
    expect(<Image
      url="http://via.placeholder.com/1600x900/CDECFF"
    />).toBeDefined();
  });
  it('renders defaults correctly', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
      />,
    )).toMatchSnapshot();
  });
  it('renders alt text when passed', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
        alt="horseback rider"
      />,
    )).toMatchSnapshot();
  });
  it('renders an object of images correctly', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
        images={{
          desktop: [
            'http://via.placeholder.com/2400x1350/CDECFF',
            'http://via.placeholder.com/1600x900/CDECFF',
          ],
          mobile: [
            'http://via.placeholder.com/800x450/CDECFF',
            'http://via.placeholder.com/800x450/CDECFF',
          ],
          tablet: [
            'http://via.placeholder.com/1600x900/CDECFF',
            'http://via.placeholder.com/1600x900/CDECFF',
          ],
        }}
      />,
    )).toMatchSnapshot();
  });
  it('renders if the object is malformed', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
        images={{
          desktop: [],
          mobile: [],
          tablet: [],
        }}
      />,
    )).toMatchSnapshot();
  });
  it('renders a Shadow prop', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
        shadow={1}
      />,
    )).toMatchSnapshot();
  });
  it('renders a Radius prop correctly', () => {
    expect(shallow(
      <Image
        url="http://via.placeholder.com/1600x900/CDECFF"
        radius={3}
      />,
    )).toMatchSnapshot();
  });
});
