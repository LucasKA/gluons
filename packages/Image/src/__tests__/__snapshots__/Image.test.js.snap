// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders a Radius prop correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="blank figure"
    images={null}
    radius={3}
    shadow={null}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        null,
        <img
          alt="blank figure"
          className="br3 shadow-null w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "blank figure",
          "className": "br3 shadow-null w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          null,
          <img
            alt="blank figure"
            className="br3 shadow-null w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blank figure",
            "className": "br3 shadow-null w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Image /> renders a Shadow prop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="blank figure"
    images={null}
    radius={0}
    shadow={1}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        null,
        <img
          alt="blank figure"
          className="br0 shadow-1 w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "blank figure",
          "className": "br0 shadow-1 w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          null,
          <img
            alt="blank figure"
            className="br0 shadow-1 w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blank figure",
            "className": "br0 shadow-1 w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Image /> renders alt text when passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="horseback rider"
    images={null}
    radius={0}
    shadow={null}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        null,
        <img
          alt="horseback rider"
          className="br0 shadow-null w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "horseback rider",
          "className": "br0 shadow-null w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          null,
          <img
            alt="horseback rider"
            className="br0 shadow-null w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "horseback rider",
            "className": "br0 shadow-null w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Image /> renders an object of images correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="blank figure"
    images={
      Object {
        "desktop": Array [
          "http://via.placeholder.com/2400x1350/CDECFF",
          "http://via.placeholder.com/1600x900/CDECFF",
        ],
        "mobile": Array [
          "http://via.placeholder.com/800x450/CDECFF",
          "http://via.placeholder.com/800x450/CDECFF",
        ],
        "tablet": Array [
          "http://via.placeholder.com/1600x900/CDECFF",
          "http://via.placeholder.com/1600x900/CDECFF",
        ],
      }
    }
    radius={0}
    shadow={null}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <source
          media="(min-width: 60em)"
          srcSet="
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x"
        />,
        <source
          media="(min-width: 30em) and (max-width: 60em)"
          srcSet="
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x"
        />,
        <source
          media="(max-width: 30em)"
          srcSet="
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 1x,
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 2x"
        />,
        <img
          alt="blank figure"
          className="br0 shadow-null w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/2400x1350/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 60em)",
          "srcSet": "
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/1600x900/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 30em) and (max-width: 60em)",
          "srcSet": "
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/800x450/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(max-width: 30em)",
          "srcSet": "
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 1x,
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "blank figure",
          "className": "br0 shadow-null w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <source
            media="(min-width: 60em)"
            srcSet="
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x"
          />,
          <source
            media="(min-width: 30em) and (max-width: 60em)"
            srcSet="
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x"
          />,
          <source
            media="(max-width: 30em)"
            srcSet="
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 1x,
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 2x"
          />,
          <img
            alt="blank figure"
            className="br0 shadow-null w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/2400x1350/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 60em)",
            "srcSet": "
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/2400x1350/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/1600x900/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 30em) and (max-width: 60em)",
            "srcSet": "
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 1x,
            http://via.placeholder.com/1600x900/CDECFF,http://via.placeholder.com/1600x900/CDECFF 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/800x450/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(max-width: 30em)",
            "srcSet": "
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 1x,
          http://via.placeholder.com/800x450/CDECFF,http://via.placeholder.com/800x450/CDECFF 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blank figure",
            "className": "br0 shadow-null w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Image /> renders defaults correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="blank figure"
    images={null}
    radius={0}
    shadow={null}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        null,
        <img
          alt="blank figure"
          className="br0 shadow-null w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "blank figure",
          "className": "br0 shadow-null w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          null,
          <img
            alt="blank figure"
            className="br0 shadow-null w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blank figure",
            "className": "br0 shadow-null w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Image /> renders if the object is malformed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="blank figure"
    images={
      Object {
        "desktop": Array [],
        "mobile": Array [],
        "tablet": Array [],
      }
    }
    radius={0}
    shadow={null}
    url="http://via.placeholder.com/1600x900/CDECFF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <source
          media="(min-width: 60em)"
          srcSet="
            null 1x,
            null 2x"
        />,
        <source
          media="(min-width: 30em) and (max-width: 60em)"
          srcSet="
            null 1x,
            null 2x"
        />,
        <source
          media="(max-width: 30em)"
          srcSet="
          null 1x,
          null 2x"
        />,
        <img
          alt="blank figure"
          className="br0 shadow-null w-100"
          src="http://via.placeholder.com/1600x900/CDECFF"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/1600x900/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 60em)",
          "srcSet": "
            null 1x,
            null 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/1600x900/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 30em) and (max-width: 60em)",
          "srcSet": "
            null 1x,
            null 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": "http://via.placeholder.com/1600x900/CDECFF",
        "nodeType": "host",
        "props": Object {
          "media": "(max-width: 30em)",
          "srcSet": "
          null 1x,
          null 2x",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "blank figure",
          "className": "br0 shadow-null w-100",
          "src": "http://via.placeholder.com/1600x900/CDECFF",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <source
            media="(min-width: 60em)"
            srcSet="
            null 1x,
            null 2x"
          />,
          <source
            media="(min-width: 30em) and (max-width: 60em)"
            srcSet="
            null 1x,
            null 2x"
          />,
          <source
            media="(max-width: 30em)"
            srcSet="
          null 1x,
          null 2x"
          />,
          <img
            alt="blank figure"
            className="br0 shadow-null w-100"
            src="http://via.placeholder.com/1600x900/CDECFF"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/1600x900/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 60em)",
            "srcSet": "
            null 1x,
            null 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/1600x900/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 30em) and (max-width: 60em)",
            "srcSet": "
            null 1x,
            null 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": "http://via.placeholder.com/1600x900/CDECFF",
          "nodeType": "host",
          "props": Object {
            "media": "(max-width: 30em)",
            "srcSet": "
          null 1x,
          null 2x",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blank figure",
            "className": "br0 shadow-null w-100",
            "src": "http://via.placeholder.com/1600x900/CDECFF",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
